app/src/org/commcare/models/database/user/UserDbUpgradeUtils.java

```diff
@@ -198,4 +200,28 @@ protected static void migrateFormRecordsToV3(Context c, SQLiteDatabase db) {
             newStorage.write(newRecord);
         }
     }
+
+    protected static void addAppIdColumnToEntityCacheTable(SQLiteDatabase db) {
+        // Drop the existing table and recreate using current definition
+        db.execSQL("DROP TABLE IF EXISTS " + EntityStorageCache.TABLE_NAME);
+        db.execSQL(EntityStorageCache.getTableDefinition());
+    }
+
+    protected static void addInterruptedFieldToSessionStateDescriptors(Context c, SQLiteDatabase db) {
+        SqlStorage<SessionStateDescriptorV1> oldStorage = new SqlStorage<>(
+                SessionStateDescriptor.STORAGE_KEY,
+                SessionStateDescriptorV1.class,
+                new ConcreteAndroidDbHelper(c, db));
+
+        SqlStorage<SessionStateDescriptor> newStorage = new SqlStorage<>(
+                SessionStateDescriptor.STORAGE_KEY,
+                SessionStateDescriptor.class,
+                new ConcreteAndroidDbHelper(c, db));
+
+        for (SessionStateDescriptorV1 ssd : oldStorage) {
```

Comment:
@amstone326 

Important note: Within this file. _all SSD references_ before this migration occurs (which is all of them for now) need to be on SSDV1, otherwise a Db update from, say, v5 to v22 could try to load the SSD it's manipulating as an SSDV2 and crash.

Reply:
@ctsims Very good catch, thank you.

Reply:
..Ah, I just realized your other suggestion makes this irrelevant. I think that is a better way to go so I'll do that refactor now